### PR TITLE
Return to home

### PR DESCRIPTION
**Pull request by ReginPotter**
# Description

When the button is pressed on The Controller the code should stop algaecide immediately. After the button is pressed and something is detected from the left or right ultrasonic sensors movement should also stop. In order to resume algaecide dispensing and movement the start/stop movement and start/stop algaecide buttons must be pressed.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A
  - [x] Start PEAT and The Controller
  - [x] Press the `Start/Stop Movement` button (Green) on The Controller
    - If the front left or right ultrasonic sensors detect anything right now it will just turn as normal
  - [x] Press the `Return to Home` button (Yellow) on The Controller
  - [x] Make the front left or right ultrasonic sensors detect something (i.e. a hand, paper, etc.)
    - This will force movement to stop

- [x] Test B
  - [x] Start PEAT and The Controller
  - [x] Press the `Start/Stop Algaecide` button (Blue) on The Controller
    - The algaecide dispensing should start as normal
  - [x] Press the `Return to Home` button (Yellow) on The Controller
    - This will force algaecide dispensing to stop

- [x] Test C
  - [x] Start PEAT and The Controller
  - [x] Press the `Start/Stop Algaecide` button (Blue) on The Controller
    - The algaecide dispensing should start as normal
  - [x] Press the `Start/Stop Movement` button (Green) on The Controller
    - If the front left or right ultrasonic sensors detect anything right now it will just turn as normal
  - [x] Press the `Return to Home` button (Yellow) on The Controller
    - this will force algaecide dispensing to stop
  - [x] Make the front left or right ultrasonic sensors detect something (i.e. a hand, a paper, etc.)
    - This will force movement to stop

# Checklist:

- [X] My code follows the style guidelines of this project (Pep 8)
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (if applicable)
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: PEAT/utils/bg_files/movement.py
Comment (toomb1r) [inline PEAT/utils/bg_files/movement.py:1]: What is the reason for the writing to a file? Is it so we can test timing when PEAT is not ssh into?

Comment (ReginPotter) [reply PEAT/utils/bg_files/movement.py:1]: The reasoning is that we have two processes running at the same time and they cannot talk together. This is the only way we can get them to talk together. I discussed this with the group yesterday and we determined using GPIO pins to solve this issue but today we found out that this proposed approach cannot work. We decided to go with File IO instead.

File: PEAT/utils/algaecide.py
[the remark this answers is not in the document]
Comment (ReginPotter) [reply PEAT/utils/algaecide.py:1]: We are waiting to remove print statements until everything is merged

**Review by L30N1DAS — APPROVED** [review]
Works in line with the modified goals. Will clean up later

**Pull request merged**